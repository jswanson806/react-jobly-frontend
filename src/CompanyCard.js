import React from "react";
import './CompanyCard.css';

/**
 * CompanyCard Component
 *
 * Renders a card displaying the details of a company, including its name and description.
 *
 * Props:
 * - company: Object containing the company details, including `name` and `description`.
 */

const CompanyCard = ({company}) => {

    return (
        <div className="Company-card" data-testid="Company-card">
            <div className="Company-heading" data-testid="Company-heading">
                <h4 className="Company-title" data-testid="Company-title">{company.name}</h4>
            </div>
            <div className="Company-description" data-testid="Company-description">
                {company.description}
            </div>
        </div>
    );
};

export default CompanyCard;