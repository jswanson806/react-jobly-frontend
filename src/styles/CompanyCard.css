.Company-card {
  margin-top: 0;
  margin-bottom: -2em;
  max-width: 600px;
  padding: 2em;
  background-color: #F5E7E1;
  border-radius: 10px;
  box-shadow: 0 8px 20px rgba(0, 0, 0, 0.15);
  font-family: 'Raleway', sans-serif;
  }

  .Company-title {
    margin-left: 1em;
    color: #0c1424;
  }
  
  .Company-description {
    margin-left: 1em;
    font-size: 1rem;
    font-weight: 500;
    color: #3c4f76;
  }

  a:-webkit-any-link {
    text-decoration: none;
  }
