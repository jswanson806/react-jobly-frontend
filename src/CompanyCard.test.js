import React from 'react';
import { act, screen, render, getByTestId} from '@testing-library/react';
import CompanyCard from './CompanyCard.js';
import '@testing-library/jest-dom';


const company = {
    handle: "test-company-1",
    name: "Test Company 1",
    num_employees: 9999,
    description: "Test description for Test Company 1",
}

describe('CompanyCard smoke and snapshot', () => {

    it('should render without crashing', async () => {
        await act(async () => {render(
            <CompanyCard company={company} />);
        });
    });

    it('should match the snapshot', async () => {
        let asFragment;
        await act(async () => {const {asFragment: fragment} = render(
            <CompanyCard company={company} />
            );
            asFragment = fragment;
        });
        
        expect(asFragment()).toMatchSnapshot();
    });

});

describe('Rendering with correct data', () => {

    it('should contain correct data from prop', async () => {
        await act(async () => {render(
            <CompanyCard company={company} />);
        });
        
        const title = screen.getByTestId("Company-title");
        const description = screen.getByTestId("Company-description");
        expect(title).toBeInTheDocument();
        expect(title).toHaveTextContent("Test Company 1");
        expect(description).toBeInTheDocument();
        expect(description).toHaveTextContent("Test description for Test Company 1");

    });

});
