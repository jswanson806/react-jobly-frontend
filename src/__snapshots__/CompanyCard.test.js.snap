// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyCard smoke and snapshot should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="Company-card"
    data-testid="Company-card"
  >
    <div
      class="Company-heading"
      data-testid="Company-heading"
    >
      <h4
        class="Company-title"
        data-testid="Company-title"
      >
        Test Company 1
      </h4>
    </div>
    <div
      class="Company-description"
      data-testid="Company-description"
    >
      Test description for Test Company 1
    </div>
  </div>
</DocumentFragment>
`;
